src/zope/container/contained.py

```diff
@@ -827,9 +824,9 @@ def chooseName(self, name, object):
         # convert to unicode and remove characters that checkName does not allow
         if isinstance(name, bytes):
             name = name.decode()
-        if not isinstance(name, unicode):
+        if not isinstance(name, text_type):
             try:
-                name = unicode(name)
+                name = text_type(name)
             except:
```

Comment:
The bare except was added back for 3.11.1 in 2009-ish. There is a test case that has `__unicode__` raise `Exception`. https://github.com/zopefoundation/zope.container/commit/54972c75936829e47ef8d35aeec3e1639a12746d

Maybe `except Exception` is better so we don't catch things like KeyboardInterrupt and such, though.